streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKTableJoin.java

```diff
@@ -32,18 +36,28 @@ class KStreamKTableJoin<K, V1, V2, VOut> implements ProcessorSupplier<K, V1, K,
     private final boolean leftJoin;
     private final Optional<Duration> gracePeriod;
     private final Optional<String> storeName;
-
+    private final Optional<StoreBuilder<?>> storeBuilder;
 
     KStreamKTableJoin(final KTableValueGetterSupplier<K, V2> valueGetterSupplier,
                       final ValueJoinerWithKey<? super K, ? super V1, ? super V2, VOut> joiner,
                       final boolean leftJoin,
                       final Optional<Duration> gracePeriod,
-                      final Optional<String> storeName) {
+                      final Optional<StoreBuilder<?>> storeBuilder) {
         this.valueGetterSupplier = valueGetterSupplier;
         this.joiner = joiner;
         this.leftJoin = leftJoin;
         this.gracePeriod = gracePeriod;
-        this.storeName = storeName;
+        this.storeName = storeBuilder.map(StoreBuilder::name);
+        this.storeBuilder = storeBuilder;
+    }
+
+    @Override
+    public Set<StoreBuilder<?>> stores() {
```

Comment:
I think `stores()` is only ever called once, so it's probably fine doing it here, but just wondering why you chose to do this here instead of in the constructor given it always returns the same thing?

Reply:
sorry, can you elaborate on what you mean by "this"? Like wrapping it in the Collections.singleton to convert it to a Set<StoreBuilder>?

Reply:
I mean instead of storing the `Optional<StoreBuilder>` we can store `Set<StoreBuilder>` and just run this same code here you have in `stores` but do it in the constructor

Reply:
gotcha, yeah we can do that